Frontend/RestaurantDashboard/src/pages/MenuList.tsx

```diff
@@ -137,26 +146,32 @@ const MenuList = () => {
                   <span className="text-sm text-gray-700">Disponível:</span>
                   <Switch
                     checked={item.available}
-                    onChange={() => toggleAvailability(idx)}
-                    className={`${
-                      item.available ? "bg-green-500" : "bg-gray-300"
-                    } relative inline-flex h-6 w-11 items-center rounded-full transition-colors`}
+                    onChange={() => toggleAvailability(item.id - 1)} // Alterado para garantir o índice correto
```

Comment:
Não precisa subtrair já que o id é uma string

```suggestion
                    onChange={() => toggleAvailability(item.id)}
```